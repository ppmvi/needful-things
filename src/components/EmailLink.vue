<template>
  <a
    href=""
    aria-label="E-Mail senden"
    @click.prevent="sendMail"
  >
    <span v-if="!$slots.default">
      {{ splittedEmail[0] }}<span class="hidden" />(at)<span class="hidden" />{{ splittedEmail[1] }}
    </span>
    <slot v-else></slot>
  </a>
</template>

<script lang="ts">
  import { Vue, Component, Prop } from 'vue-property-decorator';

  @Component
  export default class EmailLink extends Vue {
    @Prop({ required: true }) email!: string;

    get splittedEmail() {
      return this.email.split('@');
    }

    sendMail() {
      window.location.href = `mailto:${this.email}`;
    }
  }
</script>
