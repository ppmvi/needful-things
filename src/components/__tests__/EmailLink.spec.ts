import { mount, createLocalVue } from '@vue/test-utils';
import EmailLink from '../EmailLink.vue';

delete window.location;
Object.defineProperty(window, 'location', {
  value: {
    href: '',
  },
  writable: true,
});

const localVue = createLocalVue();

function wrapperFactory() {
  return mount(EmailLink, {
    localVue,
    propsData: {
      email: 'test@test.de',
    },
  });
}

describe('EmailLink.vue', () => {
  test('renders correctly', () => {
    const wrapper = wrapperFactory();
    expect(wrapper.element).toMatchSnapshot();
  });

  test('component is a vue instance', () => {
    const wrapper = wrapperFactory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('email should be splitted by @ symbol', () => {
    const wrapper = wrapperFactory();
    expect((wrapper.vm as any).splittedEmail).toEqual(['test', 'test.de']);
  });

  test('click on "a" tag should set window.location.href to mailto', () => {
    const wrapper = wrapperFactory();
    wrapper.find('a').trigger('click');
    expect(window.location.href).toBe('mailto:test@test.de');
  });
});
