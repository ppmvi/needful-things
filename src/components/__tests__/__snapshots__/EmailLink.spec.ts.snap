// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailLink.vue renders correctly 1`] = `
<a
  aria-label="E-Mail senden"
  href=""
>
  <span>
    
    test
    <span
      class="hidden"
    />
    (at)
    <span
      class="hidden"
    />
    test.de
  
  </span>
</a>
`;
